skycoin/skywire
# Dmsg delete entry on shutdown

Did you run `make format && make check`?

Fixes #	

 Changes:	
- Added `dmsgC.Close()` in `initDmsg` in modules
- Removed `go dmsgC.Serve(ctx)` from `NewTransportListener` as it was being called twice.

How to test this PR:
1. Checkout dmsg repo on PR [#110](https://github.com/skycoin/dmsg/pull/110).
2. In skywire uncomment `replace github.com/skycoin/dmsg => ../dmsg` in go.mod .
3. Run `make dep`
4. In `skywire-services` make `log:"debug"` in `integration/generic/visorA.json`
5. Start integration env in `skywire-services`
6. Run in different terminal
```bash
$ redis-cli
> get <PK-OF-VISORA>
```
7. Wait for 5 mins and check again. It should still be there.
8. Close visor A with `Ctrl+C`
9. Check for the pk again in redis. The entry should be gone.

This can be tested without integration env on `http://dmsg-discovery.myownsky.xyz`.
